### ISSUE TITLE
uninitialized constant Async::HTTP::URLEndpoint

### ISSUE THREAD

**Issue by senid231** [socketry/async-http-faraday#3]
`async-http` version `0.48.1`
`async-http-faraday` version `0.3.0`

```
"<NameError>: uninitialized constant Async::HTTP::URLEndpoint"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/async-http-faraday-0.3.0/lib/async/http/faraday/adapter.rb:45:in `endpoints_for'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/async-http-faraday-0.3.0/lib/async/http/faraday/adapter.rb:32:in `each'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/async-http-faraday-0.3.0/lib/async/http/faraday/adapter.rb:32:in `to_a'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/async-http-faraday-0.3.0/lib/async/http/faraday/adapter.rb:32:in `call'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/json_api_client-1.6.0/lib/json_api_client/middleware/parse_json.rb:6:in `call'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/json_api_client-1.6.0/lib/json_api_client/middleware/status.rb:5:in `call'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/json_api_client-1.6.0/lib/json_api_client/middleware/json_request.rb:9:in `call'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/faraday_middleware-0.12.2/lib/faraday_middleware/request/encode_json.rb:24:in `call'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/faraday-0.15.4/lib/faraday/rack_builder.rb:143:in `build_response'"
"/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/faraday-0.15.4/lib/faraday/connection.rb:387:in `run_request'"
```

I think it should be `Async::HTTP::Endpoint`
looks like last version(s) of `async-http-faraday` does not support last version of `async-http`

**Comment by senid231** [comment]
also `reason` property is missing in async http response object

```
<NoMethodError>: undefined method `reason' for #<Async::HTTP::Protocol::HTTP1::Response:0x00005605d5b49d68>
/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/async-http-faraday-0.3.0/lib/async/http/faraday/adapter.rb:36:in `call'
/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/json_api_client-1.6.0/lib/json_api_client/middleware/parse_json.rb:6:in `call'
/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/json_api_client-1.6.0/lib/json_api_client/middleware/status.rb:5:in `call'
/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/json_api_client-1.6.0/lib/json_api_client/middleware/json_request.rb:9:in `call'
/home/senid/.rvm/gems/ruby-2.5.5@panel/gems/faraday_middleware-0.12.2/lib/faraday_middleware/request/encode_json.rb:24:in `call'
```

**Comment by ioquatix** [comment]
Rename `URLEndpoint` to `Endpoint`.

Reason was removed since not present in HTTP/2.

Can you make PR?

**Comment by ioquatix** [comment]
Fixed and released in v0.4.0

**Comment by senid231** [comment]
@ioquatix thanks
